terminus-ui/selection-list/src/selection-list.component.spec.ts

```diff
@@ -875,6 +875,27 @@ describe(`TsSelectionListComponent`, function() {
       expect(selected.length).toEqual(0);
     });
 
+    test(`should be able to clear all options programmatically`, fakeAsync(() => {
+      const instance = fixture.componentInstance;
+      const selected = getAllOptionInstances(fixture).filter(o => o.selected);
+      expect(selected.length).toEqual(2);
+      instance.myCtrl.setValue([]);
+      tick(1000);
+
+      const input = getSelectionListInput(fixture);
+      typeInElement(fixture.componentInstance.states[3].name.substring(0, 3), input);
+      tick(1000);
+      fixture.detectChanges();
+
+      const opt = getOptionElement(fixture, 0, 0);
+      opt.click();
+      tick(1000);
+      fixture.detectChanges();
+
+      const chips = getAllChipInstances(fixture);
```

Comment:
Out of curiosity, any reason we check the number of selected options at the beginning but check chips at the end?

Reply:
The latter one was simply trying to mock how the bug was caught. Guess we could also check chips at beginning too..